promql/functions.go

```diff
@@ -1365,9 +1365,12 @@ func funcHistogramStdDev(vals []parser.Value, _ parser.Expressions, enh *EvalNod
 				continue
 			}
 			var val float64
-			if bucket.Lower <= 0 && 0 <= bucket.Upper {
+			switch {
+			case bucket.Lower <= 0 && bucket.Upper >= 0:
```

Comment:
I think this could be restructured slightly for better readability. Something like this:
```
			if sample.H.UsesCustomBuckets() {
			// use arithmetic mean in case of NHCB.
				val = (bucket.Upper + bucket.Lower) / 2
			} else {
				if bucket.Upper >= 0 && bucket.Lower <= 0 {
					val = 0
				} else {
					val = math.Sqrt(bucket.Upper * bucket.Lower)
					if bucket.Lower < 0 {
						val = -val
					}
				}
			}
```
Let me know your thoughts, @beorn7.